### PR TITLE
Added a Custom Dialog Builder 🛠

### PR DESCRIPTION
**Pull request by moazelsawaf**
Hello Dear 👋🏻

In this PR I have added a **Custom Dialog Builder** to build a customized upgrader dialog 🛠

Previously there were just two default styles which are Material & Cupertino with the ability to determine whether to use `UpgradeDialogStyle.cupertino` or `UpgradeDialogStyle.material`, now I am introduction a third style which is `UpgradeDialogStyle.custom`. 📱

When you set the `UpgradeDialogStyle` to be `custom`, you have to define a `customDialogBuilder` callback which receives 3 important pieces of information about the update which are: `title`, `message`, and `releaseNotes` and returns the custom built dialog 💪🏻

**Example:** ✅

Example

### CODE REVIEW

File: example/android/build.gradle [hunk 1]
```diff
@@ -1,12 +1,12 @@
 buildscript {
-    ext.kotlin_version = '1.3.50'
+    ext.kotlin_version = '1.7.20'
```
Comment (larryaasen) [inline example/android/build.gradle:2]: Can you remove this change as it has nothing to do with the dialog change.

File: example/android/build.gradle [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
     repositories {
         google()
         jcenter()
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.5.0'
+        classpath 'com.android.tools.build:gradle:7.3.0'
```
Comment (larryaasen) [inline example/android/build.gradle:9]: Can you remove this change as it has nothing to do with the dialog change.

File: example/android/gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.2-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.5.1-all.zip
```
Comment (larryaasen) [inline example/android/gradle/wrapper/gradle-wrapper.properties:6]: Can you remove this change as it has nothing to do with the dialog change.

File: README.md [hunk 4]
```diff
@@ -98,6 +98,39 @@ You can also display a Cupertino style dialog by using the `dialogStyle` paramet
...
 ![image](screenshots/example-cupertino.png)
 
+## Custom Alert Example
+
+You can also display Custom dialog by using the `dialogStyle` parameter.
+
+```dart
+          body: UpgradeAlert(
+          upgrader: Upgrader(
+            debugDisplayAlways: true,
```
Comment (larryaasen) [inline README.md:108]: This example is nice, but it should not include this: `debugDisplayAlways: true,`.

File: lib/src/upgrader.dart [hunk 5]
```diff
@@ -82,6 +82,14 @@ class Upgrader {
   /// The upgrade dialog style. Used only on UpgradeAlert. (default: material)
   UpgradeDialogStyle dialogStyle;
 
+  /// Custom Dialog Builder for dialogStyle == DialogStyle.custom
+  Widget Function(
+    BuildContext context,
+    String title,
+    String message,
+    String? releaseNotes,
+  )? customDialogBuilder;
```
Comment (larryaasen) [inline lib/src/upgrader.dart:91]: The three buttons are missing: Ignore, Later, and Update Now.

**Comment by cipchk** [comment]
Any progress? I really need `custom` mode.

**Comment by moazelsawaf** [comment]
@cipchk 
I am just very busy at this period of time, I will try hard to complete this PR in the next few days Insha'Allah.

**Comment by Nirajn2311** [comment]
Hi @moazelsawaf any update? We also need this feature before our next major release. We can take over and finish this if you're still not able to work on this.

File: example/android/build.gradle (hunk 1 above)
Comment (moazelsawaf) [reply example/android/build.gradle:2]: Removed ✅

File: example/android/build.gradle (hunk 2 above)
Comment (moazelsawaf) [reply example/android/build.gradle:9]: Removed ✅

File: example/android/gradle/wrapper/gradle-wrapper.properties (hunk 3 above)
Comment (moazelsawaf) [reply example/android/gradle/wrapper/gradle-wrapper.properties:6]: Removed ✅

File: README.md (hunk 4 above)
Comment (moazelsawaf) [reply README.md:108]: Removed ✅

File: lib/src/upgrader.dart (hunk 5 above)
Comment (moazelsawaf) [reply lib/src/upgrader.dart:94]: Added ✅

**Comment by moazelsawaf** [comment]
I have pushed the remaining part of this feature and also applied all review requests, sorry for being late 🙏🏻

**Comment by Nirajn2311** [comment]
Hi @larryaasen , any chance you could give this another review.

**Comment by StarProxima** [comment]
Hi @larryaasen, could you check it out?

**Comment by larryaasen** [comment]
There is a new version https://pub.dev/packages/upgrader/versions/9.0.0-alpha.1 that is in test and provides a way to handle this. Look at the examples included with the package.

**Comment by larryaasen** [comment]
@moazelsawaf  Look at the latest version of this package with a latest customization options. I think it now meets your needs. If not, please open a new issue with your details.

**Pull request closed without merging**